database.rules.json

```diff
@@ -2,14 +2,16 @@
   "rules": {
     "users": {
       "$uid": {
-        ".read": "auth != null && auth.uid == $uid",
-        ".write": "auth != null && auth.uid == $uid"
+        ".read" : "auth!== null && auth.uid === $uid",
+        ".write" : "auth !== null && auth.uid === $uid"
       }
     },
-    "calendars": {
-      "$calendar_id": {
-      ".read": "auth !=null && data.child('permissionRead').hasChild(root.child('users/'+auth.uid).val())",
-      ".write": "auth !=null && data.child('permissionWrite/' + auth.uid.val()).exists()"
+    "calendars":{
+      "$calendarId": {
+        "calendarState": {
+          ".read" : "auth!== null && data.parent().child('read').hasChild(auth.uid) && data.parent().child('read').child(auth.uid).val() === true",
```

Comment:
Two things here.

First, you shouldn't ever need to do `=== true`. Think about why.

Second, I think the value of the key doesn't really matter anyway. You're not going to put in uids with values that are anything other than `true` right? What about `".read": "auth !== null && data.parent().child('read/' + auth.uid).exists()"` ?